text/0000-proc-macros.md

```diff
@@ -0,0 +1,417 @@
+- Feature Name: procedural_macros
+- Start Date: 2016-02-15
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes an evolution of Rust's procedural macro system (aka syntax
+extensions, aka compiler plugins). This RFC specifies syntax for the definition
+of procedural macros, a high-level view of their implementation in the compiler,
+and outlines how they interact with the compilation process.
+
+This RFC specifies the architecture of the procedural macro system. It relies on
+[RFC 1561](https://github.com/rust-lang/rfcs/pull/1561) which specifies the
+naming and modularisation of macros. It leaves many of the details for further
+RFCs, in particular the details of the APIs available to macro authors
+(tentatively called `libmacro`). See this [blog post](http://ncameron.org/blog/libmacro/)
```

Comment:
A half-nit-picking and half-question: `macro` is a reserved keyword, so wouldn't `extern crate macro;` and subsequent uses of `macro::...` be a syntax error? We need to put a special status to the `macro` keyword or rename the crate.